src/amber/router/context.cr

```diff
@@ -13,8 +13,9 @@ class HTTP::Server::Context
 
   def initialize(@request : HTTP::Request, @response : HTTP::Server::Response)
     router = Amber::Router::Router.instance
-    @route = router.match_by_request(@request)
     parse_params
     upgrade_request_method!
```

Comment:
The order matters here since we need parse the params first in order to check if the request method can be upgraded based on the `_method`. Then it tries to find the route for the request.